trace/data.go

```diff
@@ -151,30 +152,19 @@ func (tr *trace) prepareSpanData(ctx context.Context, request json.RawMessage, r
 		ApplicationRuntimeVersion: plugin.ApplicationRuntimeVersion,
 		ApplicationTags:           map[string]interface{}{},
 
-		TraceId:        plugin.TraceId,
-		TransactionId: plugin.TransactionId,
-		//ParentSpanId:    Only root span exist right now
+		TraceID:       span.Context.TraceID,
+		TransactionID: plugin.TransactionID,
+		ParentSpanID:  span.ParentSpanID,
 
 		DomainName:    plugin.ApplicationDomainName,
 		ClassName:     plugin.ApplicationClassName,
 		ServiceName:   plugin.FunctionName, //TODO implement it with Opentracing
-		OperationName: plugin.FunctionName, //TODO implement it with Opentracing
+		OperationName: span.OperationName,
 
-		StartTimestamp:  tr.span.startTime,
-		FinishTimestamp: tr.span.finishTime,
-		Duration:        tr.span.duration,
-		Tags:            tags,
-	}
-}
-
-func (tr *trace) prepareSpanTags(ctx context.Context, request json.RawMessage, response interface{}) map[string]interface{} {
-	tags := map[string]interface{}{}
-	// If the agent's user doesn't want to send their request and response data, hide them.
-	if !shouldHideRequest() {
-		tags[plugin.AwsLambdaInvocationRequest] = string(request)
+		StartTimestamp:  span.StartTimestamp,
+		FinishTimestamp: span.EndTimestamp,
+		Duration:        span.Duration(),
+		Tags:            span.Tags,
+		Logs:            map[string]spanLog{}, // TO DO get logs
```

Comment:
you can add logs as `logs` property in the `Span` as mentioned in the data model doc.